ndcube/mixins/ndcubeio.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+import numpy as np
+from astropy.nddata import NDIOMixin, CCDData
+from astropy.io import fits, registry
+from astropy import units as u
+from astropy.nddata.nduncertainty import (
+    StdDevUncertainty, NDUncertainty, VarianceUncertainty, InverseVariance)
+
+_known_uncertainties = (StdDevUncertainty, VarianceUncertainty, InverseVariance)
+_unc_name_to_cls = {cls.__name__: cls for cls in _known_uncertainties}
+_unc_cls_to_name = {cls: cls.__name__ for cls in _known_uncertainties}
+
+
+__all__ = ['NDCubeIOMixin']
+
+# This file is written to define a mixin class to support
+# write option in ndcube. Uses `astropy.io.registry`
+# to register the write function
+
+
+def flatten(lst):
+    """Helper function to flatten a list of dictionary,
+       resulted out of the form :
+       {Key1:Value1, {Key2:Value2, Key3:Value3}}
+
+
+    Parameters
+    ----------
+    lst : list
+        list of dictionary
+
+    Returns
+    -------
+    list
+        list of flattened list
```

Comment:
From below it looks like the output is actually a list of dictionaries.

Reply:
Same as above